sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -575,7 +575,7 @@ class CodegenContext {
     if (freshNameIds.contains(fullName)) {
       val id = freshNameIds(fullName)
       freshNameIds(fullName) = id + 1
-      s"$fullName$id"
+      s"${fullName}_$id"
     } else {
       freshNameIds += fullName -> 1
       fullName
```

Comment:
If the given name is something like `name1_1`, I think you can still produce non-unique variable name.

Reply:
I'd suggest something like `s"${fullName}_0"` at L581. It also solves the failed tests.

Reply:
If I am correct, does this still have a conflict between `a_01` wth `a_0$id` where `$id = 1`?

Reply:
Isn't changed to `s"${fullName}_$id"`? So you will get `a_0_1`.

Reply:
I think it still has a problem. for sequence `a_1`, `a`, `a`, we have duplicated name `a_1`.

We can solve this problem by always adding the postfix.

Reply:
Hmm, doesn't it be

`a_1` -> `a_1_0`
`a` -> `a_0`
`a` -> `a_1`

?

Reply:
Ah, sorry for my misunderstanding. To change line 581 would work well.

Reply:
@viirya my comment was for the current code without your change.